.github/workflows/main.yml

```diff
@@ -20,52 +22,25 @@ jobs:
       uses: microsoft/setup-msbuild@v1
 
     - name: Restore
-      run: msbuild /restore Xamarin.Forms.sln
-    
+      run: |
+        msbuild /restore Xamarin.Forms.sln
+        nuget restore Xamarin.Forms.Platform.GTK\Xamarin.Forms.Platform.GTK.csproj
+
     - name: Build
-      run: msbuild .\Xamarin.Forms.sln -p:Configuration=Release
+      run: |
+        msbuild .\Xamarin.Forms.sln -p:Configuration=${{env.CONFIGURATION}}
+        msbuild Xamarin.Forms.Platform.GTK\Xamarin.Forms.Platform.GTK.csproj /property:Configuration=${{env.CONFIGURATION}}
```

Comment:
are you sure that the first msbuild command doesn't already build the GTK project? why dont you test removing the second msbuild command to see if it's needed?